### PR TITLE
(SDI-2052) add task for metadata generation for plugins

### PR DESCRIPTION
**Pull request by nanliu**
Add PLUGINS_CATALOG to the template.
* Use github api to fetch plugins.yml
* Retrieve each repo github metadata
* Retrieve each repo .sync.yml and metadata.yml
* Fetch github issues tagged wish-list
* Generate PLUGINS_CATALOG from template

### CODE REVIEW

File: lib/pluginsync/util.rb [hunk 1]
```diff
@@ -0,0 +1,164 @@
...
+      end
+    end
+
+    ##
+    # only capitalize orgs we recognize
+
+    def self.org_capitalize(word)
+      {
+        'intelsdi-x': 'Intel',
+        'raintank': 'Raintank',
```
Comment (mjbrender) [inline lib/pluginsync/util.rb]: raintank as a company is lowercase.

File: modulesync.yml [hunk 2]
```diff
@@ -3,3 +3,6 @@ git_base: 'git@github.com:'
 namespace: intelsdi-x
 branch: pluginsync
 message: "Updates to repository from pluginsync utility"
+plugins.yml:
```
Comment (mjbrender) [inline modulesync.yml]: This is super cool 👌

File: PLUGINS_CATALOG.md [hunk 3]
```diff
@@ -0,0 +1,37 @@
+# Plugin Catalog
```
Comment (mjbrender) [inline PLUGINS_CATALOG.md]: The resulting file in the Snap repo is `PLUGIN_CATALOG.md`. Would make more sense for this name to be consistent.

File: PLUGINS_CATALOG.md [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+| Name | Maintainer | Description | CI | Download |
+|------|------------|-------------|----|----------|
+<%-
+  metadata.find_all{|p| p["type"] == type }.sort_by{|p| p["name"]}.each do |p|
+    maintainer = Pluginsync::Util.org_capitalize(p["maintainer"])
+    downloads = []
+    downloads += ["[release](#{p['github_release']})"] if p.include? "github_release"
+    downloads += p["download"]["s3_latest"].collect{|h| "[#{h.keys.first}](#{h[h.keys.first]})" } if p["download"] and p["download"]["s3_latest"]
+-%>
+| [<%= p["name"] %>](<%= p['repo_url'] %>) | <%= maintainer %> | <%= p["description"] %> | <%= Pluginsync::Util.html_list(p["badge"]) -%> | <%= Pluginsync::Util.html_list(downloads) %> |
```
Comment (mjbrender) [inline PLUGINS_CATALOG.md]: When working on https://github.com/intelsdi-x/snap/pull/1231, I liked @IRCody's idea to link to the maintainer. Could we swap that in? Thinking: 

`<%= maintainer %>](<%= p['html_url'] %>` << I think it's `html_url` here.

File: lib/pluginsync/util.rb (hunk 1 above)
Comment (nanliu) [inline lib/pluginsync/util.rb]: I removed them from this list.

File: PLUGINS_CATALOG.md (hunk 3 above)
Comment (nanliu) [inline PLUGINS_CATALOG.md]: Fixed.

File: PLUGINS_CATALOG.md (hunk 4 above)
Comment (nanliu) [inline PLUGINS_CATALOG.md]: Updated

**Comment by IRCody** [comment]
@mjbrender: Can you review @nanliu's updates and see if they're good?

File: Rakefile [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+  desc "generate pull request for plugin_metadata.json"
+  task :pull_request do
+    Pluginsync::Plugins.pull_request
+  end
+end
+
+namespace :notify do
+  desc "send a slack notification"
+  task :slack do
+    Pluginsync::Notify::Slack.message "#build-snap", "Snap packages version <https://packagecloud.io/nanliu/snap|#{@snap.pkgversion} now available.>"
```
Comment (kindermoumoute) [inline Rakefile]: Where is `{@snap.pkgversion}` defined?

File: lib/pluginsync/github.rb [hunk 6]
```diff
@@ -0,0 +1,208 @@
...
+      end
+
+      def yml_content(path, default={})
+        YAML.load(content(path))
+      rescue
+        default
+      end
+
+      def plugin_name
+        @name.match(/snap-plugin-(collector|processor|publisher)-(.*)$/)
```
Comment (kindermoumoute) [inline lib/pluginsync/github.rb]: Do we want this to work with kubesnap plugins?

Comment (mjbrender) [inline lib/pluginsync/github.rb]: @kindermoumoute nope - kubesnap project will merge into snap plugins in the longer run.

File: Rakefile (hunk 5 above)
Comment (mjbrender) [inline Rakefile]: Looks related to the bottom of `deps.sh`, but I have no freaking clue how yet 👀

**Comment by mjbrender** [comment]
@nanliu so, I broke this by changing the labeling strategy for the wish list. They're now three separate ones: 

* plugin-wishlist/collector
* plugin-wishlist/processor
* plugin-wishlist/publisher

The bad is it breaks plugins.rb behavior at the moment. The benefit is you can tag the plugin type in the resulting JSON. 

@IRCody don't merge until we're set here.

File: Rakefile (hunk 5 above)
Comment (nanliu) [inline Rakefile]: Oh, this needs to be fixed, since it's originally for package releases.

**Comment by nanliu** [comment]
@mjbrender, I added plugin type support for wishlist to the PR.

**Pull request merged**